src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -371,6 +385,9 @@ private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(DiagnosticBag dia
                         ImmutableArray.Create(location),
                         ImmutableArray.Create(parameter.GetReference()));
 
+                // Force attribute binding for diagnostics
+                typeParameter.GetAttributesBag(diagnostics);
+
```

Comment:
Consider renaming ```GrabDiagnostics``` to "GetDeclarationDiagnostics" or something similar. In general, we should make sure that diagnostics for a local function declaration never makes its way to the global (Compilation wide) diagnostics bag, which is used to cache declaration errors for regular declarations.